### PR TITLE
Improve logging in legacy megolm code

### PR DESCRIPTION
**Pull request by richvdh**
This is a series of changes to try to improve the quality of the logging within the crypto subsystem; we try to reduce the noise, but in its stead add a little more logging to try and understand what is going on with https://github.com/vector-im/element-web/issues/23269.

Suggest reviewing commit-by-commit.

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Review by alunturner — APPROVED** [review]
Looks good to me, the only notes I've added are all on the same subject and just there for consistency's sake

File: src/crypto/algorithms/megolm.ts [hunk 1]
```diff
@@ -431,25 +444,40 @@ export class MegolmEncryption extends EncryptionAlgorithm {
...
+                            .flat(1);
+
+                        if (retryDeviceList.length > 0) {
+                            this.prefixedLogger.debug(
+                                "Sharing keys (start phase 2) with devices without existing Olm sessions:",
+                                retryDeviceList,
+                            );
+                            await this.shareKeyWithDevices(session, key, payload, retryDevices, failedDevices, 30000);
+                            this.prefixedLogger.debug(
+                                `Shared keys (end phase 2) with devices without existing Olm sessions`,
```
[the remark this answers is not in the document]
Comment (richvdh) [reply src/crypto/algorithms/megolm.ts:460]:
```suggestion
                                "Shared keys (end phase 2) with devices without existing Olm sessions",
```

File: src/crypto/algorithms/megolm.ts [hunk 2]
```diff
@@ -948,24 +944,28 @@ export class MegolmEncryption extends EncryptionAlgorithm {
...
-            logger.debug(
-                `Already started preparing to encrypt for ${this.roomId} ` + `${elapsedTime} ms ago, skipping`,
+            this.prefixedLogger.debug(
+                `Already started preparing to encrypt for this room ${elapsedTime}ms ago, skipping`,
             );
             return;
         }
 
-        logger.debug(`Preparing to encrypt events for ${this.roomId}`);
+        this.prefixedLogger.debug(`Preparing to encrypt events`);
```
[the remark this answers is not in the document]
Comment (richvdh) [reply src/crypto/algorithms/megolm.ts:994]:
```suggestion
        this.prefixedLogger.debug("Preparing to encrypt events");
```

File: src/crypto/algorithms/megolm.ts [hunk 3]
```diff
@@ -994,7 +994,7 @@ export class MegolmEncryption extends EncryptionAlgorithm {
      * @returns Promise which resolves to the new event body
      */
     public async encryptMessage(room: Room, eventType: string, content: IContent): Promise<IMegolmEncryptedContent> {
-        logger.log(`Starting to encrypt event for ${this.roomId}`);
+        this.prefixedLogger.log(`Starting to encrypt event`);
```
[the remark this answers is not in the document]
Comment (richvdh) [reply src/crypto/algorithms/megolm.ts:1028]:
```suggestion
        this.prefixedLogger.log("Starting to encrypt event");
```

**Pull request merged**